### PR TITLE
Handle garbage path input

### PR DESCRIPTION
**Pull request by romanofski**
This patch fixes a problem with the path editor setting the working
directory in the file browser. Since the file browser itself is not
checking whether the working directory is valid, it is up to the caller
to verify we're not setting anything invalid.

This patch checks whether the path for the working directory exists and
otherwise shows an error.

Fixes https://github.com/purebred-mua/purebred/issues/391

### CODE REVIEW

File: src/UI/Status/Main.hs [hunk 1]
```diff
@@ -104,6 +105,9 @@ instance WithContext (ListWithLength t e) where
 instance WithContext (E.Editor Text Name) where
   renderContext _ = str . show . cursorPosition . view E.editContentsL
 
+instance WithContext (FB.FileBrowser Name) where
+  renderContext _ _ = str "filebrowser"
```
Comment (romanofski) [inline src/UI/Status/Main.hs:109]: Hm.. just realizing this little bit. The file browser widget itself has a status bar. Will have to see what makes sense to display in our status bar.

**Comment by romanofski** [comment]
I just realised that there is an inconsistency with toggling. The Brick filebrowser default toggles items with using the `Space` key, while Purebred has bound it to `*`. The tricky thing comes from re-binding it to whatever you choose, since I've just noticed that with the current setup is tricky. I think if Brick can expose access to the underlying filebrowser widget's list we would be able to use what we already have in place for our `HasToggleableList` and everything would be peachy.

**Comment by romanofski** [comment]
Went down the rabbit hole just to find out that it isn't as easy as I thought by exporting the list accessor on the file browser widget. Why? Because Brick's file browser is storing the toggled items in a separate `Set`, while we keep the state on the list item itself.
I've added an additional `Action` to toggle file browser items. It's a bit meh, but should suffice for now.

**Comment by frasertweedale** [comment]
I'm ok with it if you want to merge.  More LOC removed than added :+1: .  I don't see any showstoppers

**Pull request merged**